src/categories/categories.service.ts

```diff
@@ -0,0 +1,34 @@
+import { Injectable, NotFoundException } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+import { Category } from './category.entity';
+import { CreateCategoryDto } from './create-category.dto';
+
+@Injectable()
+export class CategoriesService {
+  constructor(
+    @InjectRepository(Category)
+    private readonly categoriesRepository: Repository<Category>,
+  ) {}
+
+  async create(createCategoryDto: CreateCategoryDto): Promise<Category> {
+    const category = new Category();
+    category.name = createCategoryDto.name;
```

Comment:
Why create & THEN mutate rather than just creating with received DTO?

Also if you extend the entities from BaseEntity, you can use ActiveRecord.

PS: Use TypeORM's create() rather than new Category()